### PR TITLE
Refactor AccountActivationMailer and SendConfirmationEmailController

### LINKED ISSUE #2193: Confirmation email code duplication
Hello,

I've noticed that the [SendConfirmationEmailController](https://github.com/flarum/core/blob/4da2994d1fab1a167406680876101d19603cc27a/src/Api/Controller/SendConfirmationEmailController.php) and [AccountActivationMailer](https://github.com/flarum/core/blob/4da2994d1fab1a167406680876101d19603cc27a/src/User/AccountActivationMailer.php) classes do almost exactly the same thing, with the only difference that they are triggered in different situations.

I think that there's space for improvement and that the code duplication could be avoided.

### PR DESCRIPTION
**Pull request by sl-kr**
**Fixes #2193**

**Changes proposed in this pull request:**

Moves the common code between [SendConfirmationEmailController](https://github.com/flarum/core/blob/master/src/Api/Controller/SendConfirmationEmailController.php) and [AccountActivationMailer](https://github.com/flarum/core/blob/master/src/User/AccountActivationMailer.php) into a trait.

**Reviewers should focus on:**

I wasn't sure what to name the trait or where to put it, but because most of the code for the trait was taken from the AccountActivationMailer class I decided to name it the AccountActivationMailerTrait and keep it in the same directory. Also, I decoupled the generateToken and getEmailData functions because I think it will make more modular and maintainable even through it may make using the trait more verbose.

**Confirmed**

- [x] Backend changes: tests are green (run `composer test`).

### CODE REVIEW

File: src/User/AccountActivationMailerTrait.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    /**
+     * Get the data that should be made available to email templates.
+     *
+     * @param User $user
+     * @param string $email
+     * @param EmailToken $token
+     * @return array
+     */
+    protected function getEmailData(User $user, $email, EmailToken $token)
```
Comment (askvortsov1) [inline src/User/AccountActivationMailerTrait.php:37]: Doesn't look like the email is actually used here

File: src/User/AccountActivationMailerTrait.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+use Flarum\Mail\Job\SendRawEmailJob;
+
+trait AccountActivationMailerTrait
+{
+    /**
+     * @param User $user
+     * @param string $email
+     * @return EmailToken
+     */
+    protected function generateToken(User $user, $email)
```
Comment (askvortsov1) [inline src/User/AccountActivationMailerTrait.php:21]: I'm leaning towards keeping it like this, with the email passed in separately to allow future flexibility if we decide to adopt multiple emails per user.

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

**Review by askvortsov1 — APPROVED** [review]
Looks good to me!

**Comment by askvortsov1** [comment]
A side effect of this would implement https://github.com/flarum/core/issues/2515, which still needs a bit more discussion before merging this.

**Review by SychO9 — APPROVED** [review]

**Pull request merged**